internal/impl/config.go

```diff
@@ -43,6 +48,29 @@ type Stage struct {
 	Command string `cue:"string" json:"command"`
 }
 
+var commitMismatchErrorShown = false
+
+func (c *Config) Packages(w io.Writer) []string {
+	if featureflag.Home.Disabled() {
+		return c.RawPackages
+	}
+	path, _ := GlobalConfigPath()
+	global, err := readConfig(filepath.Join(path, "devbox.json"))
+	if err != nil {
+		return c.RawPackages
+	}
+	if c.Nixpkgs.Commit != global.Nixpkgs.Commit && !commitMismatchErrorShown {
+		commitMismatchErrorShown = true
+		color.New(color.FgHiYellow).Fprint(w, "Warning: ")
+		fmt.Fprintln(
+			w,
+			"local and global devbox.json have different nixpkgs commits. "+
+				"Will use the local version. This may lead to version mismatch and "+
+				"nix store bloat.")
+	}
```

Comment:
The odds of hitting this warning sound pretty high. For example, if we update the default hash in devbox and then all new projects end up mismatching the global devbox.json. I don't think there's an obvious way to update the commit hash in the global debox.json either, especially if the user doesn't know where it is.

Instead, could we use the global nixpkgs commit hash for global packages and the project commit hash for project packages? That sounds pretty reasonable and then we wouldn't have to show a warning.

Reply:
@gcurtis I want to do that, but need to wait for flakes stuff to be in. We can modify the flakes file to use two different nixpkgs. Though even in that case, I would want to show some sort of warning?